### PR TITLE
Convert sdk.Int to BigDec

### PR DESCRIPTION
**Pull request by hieuvubk**
Closes: #6370 

## What is the purpose of the change

Create a func `BigDecFromSDKInt` to convert sdk.Int => BigDec instead of converting sdk.Int => sdk.Dec => BigDec
Benchmark:
```
OLD:
BenchmarkBigDecFromInt-16        2074845               579.7 ns/op           640 B/op         18 allocs/op

NEW:
BenchmarkBigDecFromInt-16        3566844               312.9 ns/op           312 B/op          9 allocs/op
```

## Testing and Verifying

*(Please pick one of the following options)*

This change is a trivial rework / code cleanup without any test coverage.

*(or)*

This change is already covered by existing tests, such as *(please describe tests)*.

*(or)*

This change added tests and can be verified as follows:

*(example:)*
  - *Added unit test that validates ...*
  - *Added integration tests for end-to-end deployment with ...*
  - *Extended integration test for ...*
  - *Manually verified the change by ...*

## Documentation and Release Note

  - [ ] Does this pull request introduce a new feature or user-facing behavior changes?
  - [ ] Changelog entry added to `Unreleased` section of `CHANGELOG.md`?

Where is the change documented? 
  - [ ] Specification (`x/{module}/README.md`)
  - [ ] Osmosis documentation site
  - [ ] Code comments?
  - [ ] N/A

### CODE REVIEW

File: osmomath/decimal.go
Comment (p0mvn) [inline osmomath/decimal.go:589]: Note that this is not mutative. See that `BigInt()` reinitializes the underlying buffer.


Let's still keep this, please. However, I'm also wondering if we can make a mutative version that would reuse the input's buffer?

**Review by p0mvn — CHANGES_REQUESTED** [review]
Thanks for the change, please see comment

File: osmomath/decimal.go
Comment (pysel) [inline osmomath/decimal.go:585]: nit: I think we do not reference sdk anymore, Int is defined in osmomath package
```suggestion
// BigDecFromSDKInt returns the BigDec representation of an Int.
```

**Review by pysel — COMMENTED** [review]
thanks! left a small nit, lgtm when Roman's comment is addressed

File: osmomath/decimal.go
Comment (hieuvubk) [reply osmomath/decimal.go:589]: Yeah I see, currently it's hard to access `Int.i` directly without `BigInt()`

Comment (p0mvn) [reply osmomath/decimal.go:589]: Could you make an issue to consider this in the future and potentially talk to the SDK team about exposing such API?

**Comment by p0mvn** [comment]
@hieuvubk  please add a changelog entry and create an issue. Thanks!

**Comment by p0mvn** [comment]
Marking as draft until the requests are resolved

File: osmomath/decimal.go
Comment (hieuvubk) [reply osmomath/decimal.go:589]: Created an issue for tracking [#17772](https://github.com/cosmos/cosmos-sdk/issues/17772)

**Comment by hieuvubk** [comment]
> @hieuvubk please add a changelog entry and create an issue. Thanks!

Added!
Tracking issue: [#17772](https://github.com/cosmos/cosmos-sdk/issues/17772)

File: osmomath/decimal.go
Comment (hieuvubk) [reply osmomath/decimal.go:589]: I tried returning Int.i in my local, performance is improved but the potential risks are unclear

**Comment by p0mvn** [comment]
Thanks @hieuvubk 

There are more TODOs left related to Dec() functions with no reallocations:
https://github.com/osmosis-labs/osmosis/blob/1f21fc37269b8ea14719dfd90bfaa0bb923b2ac9/x/concentrated-liquidity/math/math.go#L29

Are you planning on working on that as well? @hieuvubk

**Review by p0mvn — APPROVED** [review]

**Comment by hieuvubk** [comment]
> Thanks @hieuvubk
> 
> There are more TODOs left related to Dec() functions with no reallocations:
> 
> https://github.com/osmosis-labs/osmosis/blob/1f21fc37269b8ea14719dfd90bfaa0bb923b2ac9/x/concentrated-liquidity/math/math.go#L29
> 
> Are you planning on working on that as well? @hieuvubk



> Thanks @hieuvubk
> 
> There are more TODOs left related to Dec() functions with no reallocations:
> 
> https://github.com/osmosis-labs/osmosis/blob/1f21fc37269b8ea14719dfd90bfaa0bb923b2ac9/x/concentrated-liquidity/math/math.go#L29
> 
> Are you planning on working on that as well? @hieuvubk

I thought #6261  should resolve `Dec()`, also Im working on `QuoRoundUp` & `QuoTruncate`

**Comment by p0mvn** [comment]
@hieuvubk could you investigate the e2e failure please? I tried rerunning it a few times, and it didn't work. From logs, unclear what the issue is so might require investigations

**Comment by hieuvubk** [comment]
> @hieuvubk could you investigate the e2e failure please? I tried rerunning it a few times, and it didn't work. From logs, unclear what the issue is so might require investigations

I think it's due to a version conflict of `osmomath` because `go.mod` is taking the version of `osmomath` as `v0.0.7-0.20230911120014-b14342e08daf` while my change hasn't been tagged yet.

**Comment by p0mvn** [comment]
> > @hieuvubk could you investigate the e2e failure please? I tried rerunning it a few times, and it didn't work. From logs, unclear what the issue is so might require investigations
> 
> I think it's due to a version conflict of `osmomath` because `go.mod` is taking the version of `osmomath` as `v0.0.7-0.20230911120014-b14342e08daf` while my change hasn't been tagged yet.

Please run:
```
go get github.com/osmosis-labs/osmosis/osmomath@4225d5f
```

Should update the tag to the one taken from a commit on your branch. You should see updates in `go.mod` and `go.sum`. Please also run `go work sync` after and make sure that there are no errors

**Comment by hieuvubk** [comment]
> Please run:
> 
> ```
> go get github.com/osmosis-labs/osmosis/osmomath@4225d5f
> ```
> 
> Should update the tag to the one taken from a commit on your branch. You should see updates in `go.mod` and `go.sum`. Please also run `go work sync` after and make sure that there are no errors
Thanks @p0mvn it works!

**Comment by hieuvubk** [comment]
thanks @p0mvn it works!

**Review by p0mvn — APPROVED** [review]

File: x/concentrated-liquidity/math/math.go
Comment (p0mvn) [inline x/concentrated-liquidity/math/math.go:21]: Let's keep these comments please.

Do you mint creating an issue for this osmosis side and linking it to https://github.com/cosmos/cosmos-sdk/issues/17772?

Thanks!

**Review by p0mvn — CHANGES_REQUESTED** [review]
One more request before merge please @hieuvubk

File: x/concentrated-liquidity/math/math.go
Comment (hieuvubk) [reply x/concentrated-liquidity/math/math.go]: Sure!

Comment (hieuvubk) [reply x/concentrated-liquidity/math/math.go]: #6429

File: CHANGELOG.md
Comment (p0mvn) [inline CHANGELOG.md:68]:
```suggestion
```

**Review by p0mvn — APPROVED** [review]

**Pull request merged**
